esp/services/ws_store/ws_storeService.cpp

```diff
@@ -271,6 +271,12 @@ bool CwsstoreEx::onListKeys(IEspContext &context, IEspListKeysRequest &req, IEsp
 
     StringArray keys;
     m_storeProvider->fetchKeySet(keys, storename, ns, secuser.get(), !req.getUserSpecific());
+    for (unsigned i = 0; i < keys.length(); i++)
+    {
+        StringBuffer decoded;
+        decodePtreeName(decoded, keys.item(i));
```

Comment:
These have already been decoded within the function - so they will currently be decoded twice.   That could potentially cause problems with some pathological key names.